core/src/test/java/org/elasticsearch/indices/recovery/RecoveryStateTest.java

```diff
@@ -154,7 +154,7 @@ public void run() {
         if (randomBoolean()) {
             timer.stop();
             assertThat(timer.stopTime(), greaterThanOrEqualTo(timer.startTime()));
-            assertThat(timer.time(), equalTo(timer.stopTime() - timer.startTime()));
+            assertThat(timer.time(), lessThanOrEqualTo(timer.stopTime() - timer.startTime()));
```

Comment:
I see what your are saying but  I don’t think we can rely on this. The nanoTime() is not guaranteed to actually have nano precision (just resolution). it is only guaranteed to never go back… 



> On 29 Jun 2015, at 10:39, Masaru Hasegawa <notifications@github.com> wrote:
> 
> In core/src/test/java/org/elasticsearch/indices/recovery/RecoveryStateTest.java:
> 
> > @@ -154,7 +154,7 @@ public void run() {
> >          if (randomBoolean()) {
> >              timer.stop();
> >              assertThat(timer.stopTime(), greaterThanOrEqualTo(timer.startTime()));
> > -            assertThat(timer.time(), equalTo(timer.stopTime() - timer.startTime()));
> > +            assertThat(timer.time(), lessThanOrEqualTo(timer.stopTime() - timer.startTime()));
> 
> I think lessThanOrEqualTo is correct. (because it's rounded down to nearest decimal value)
> 
> If we use nano seconds, when start time is 1ns and stop time is 1000000ns (1ms), time() would be 99999ns but it's 0ms because of TimeValue.nsecToMSec.
> But if we use milliseconds, above becomes 1ms - 0ms = 1ms. In this case, time() < stop() - start().
> When start time is 1ns (0ms) and stop time is 1999999ns (1ms), it's 1999998ns but time() will be 1ms and stop() - start() = 1ms.
> 
> That's said, I like time() > 0 since it makes it simpler.
> 
> —
> Reply to this email directly or view it on GitHub.
>